pkg/providers/vsphere/vmprovider_vm.go

```diff
@@ -734,49 +737,37 @@ func (vs *vSphereVMProvider) createVirtualMachineAsync(
 		chanErr <- ErrCreate
 	}
 
-	_, k8sErr := controllerutil.CreateOrPatch(
-		ctx,
-		vs.k8sClient,
-		ctx.VM,
-		func() error {
+	objPatch := ctrlclient.MergeFrom(ctx.VM.DeepCopy())
```

Comment:
@bryanv I went ahead and got rid of this anachronistic use of `CreateOrPatch` about which you warned me in the past. Now it just does a pure patch to ensure we do not hit weirdness.